### ISSUE TITLE
which version of tendermint i can use for solve this issue.

### ISSUE THREAD

**Issue by puneetsingh166** [tendermint/tendermint#7032]
hey can anyone help me regarding this error. this error occurring when running ``` go mod tidy ``` 
 ``` go: finding module for package github.com/tendermint/tendermint/rpc/lib/typesgo: finding module for package github.com/tendermint/tendermint/libs/common
github.com/you/my-load-tester/cmd/my-load-tester imports
        github.com/interchainio/tm-load-test/pkg/loadtest imports
        github.com/tendermint/tendermint/libs/common: module github.com/tendermint/tendermint@latest found (v0.34.13, replaced by github.com/tendermint/tendermint@v0.34.13), but does not contain package github.com/tendermint/tendermint/libs/common
github.com/you/my-load-tester/cmd/my-load-tester imports
        github.com/interchainio/tm-load-test/pkg/loadtest imports
        github.com/tendermint/tendermint/rpc/lib/types: module github.com/tendermint/tendermint@latest found (v0.34.13, replaced by github.com/tendermint/tendermint@v0.34.13), but does not contain package github.com/tendermint/tendermint/rpc/lib/types ```

**Comment by marbar3778** [comment]
It seems you were using an old version of tendermint. TO update to the latest version you will have to fix the imports.

**Comment by puneetsingh166** [comment]
HI @marbar3778 Thanks for response hoping i am using updated version of tendermint 0.34.13

here my go.mod file 
``` module github.com/you/my-load-tester

go 1.17

require github.com/cosmos/cosmos-sdk v0.44.0

require (
	github.com/kr/text v0.2.0 // indirect
	github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e // indirect
	github.com/onsi/ginkgo v1.16.4 // indirect
	github.com/onsi/gomega v1.13.0 // indirect
	gopkg.in/check.v1 v1.0.0-20200227125254-8fa46927fb4f // indirect
)

require (
	filippo.io/edwards25519 v1.0.0-beta.2 // indirect
	github.com/99designs/keyring v1.1.6 // indirect
	github.com/ChainSafe/go-schnorrkel v0.0.0-20200405005733-88cbf1b4c40d // indirect
	github.com/DataDog/zstd v1.4.5 // indirect
	github.com/armon/go-metrics v0.3.9 // indirect
	github.com/beorn7/perks v1.0.1 // indirect
	github.com/bgentry/speakeasy v0.1.0 // indirect
	github.com/btcsuite/btcd v0.22.0-beta // indirect
	github.com/cespare/xxhash v1.1.0 // indirect
	github.com/cespare/xxhash/v2 v2.1.1 // indirect
	github.com/confio/ics23/go v0.6.6 // indirect
	github.com/cosmos/go-bip39 v1.0.0 // indirect
	github.com/cosmos/iavl v0.17.1 // indirect
	github.com/cosmos/ledger-cosmos-go v0.11.1 // indirect
	github.com/cosmos/ledger-go v0.9.2 // indirect
	github.com/danieljoos/wincred v1.0.2 // indirect
	github.com/davecgh/go-spew v1.1.1 // indirect
	github.com/dgraph-io/badger/v2 v2.2007.2 // indirect
	github.com/dgraph-io/ristretto v0.0.3 // indirect
	github.com/dgryski/go-farm v0.0.0-20200201041132-a6ae2369ad13 // indirect
	github.com/dustin/go-humanize v1.0.0 // indirect
	github.com/dvsekhvalnov/jose2go v0.0.0-20200901110807-248326c1351b // indirect
	github.com/enigmampc/btcutil v1.0.3-0.20200723161021-e2fb6adb2a25 // indirect
	github.com/felixge/httpsnoop v1.0.1 // indirect
	github.com/fsnotify/fsnotify v1.5.1 // indirect
	github.com/go-kit/kit v0.10.0 // indirect
	github.com/go-logfmt/logfmt v0.5.0 // indirect
	github.com/godbus/dbus v0.0.0-20190726142602-4481cbc300e2 // indirect
	github.com/gogo/gateway v1.1.0 // indirect
	github.com/gogo/protobuf v1.3.3 // indirect
	github.com/golang/protobuf v1.5.2 // indirect
	github.com/golang/snappy v0.0.3 // indirect
	github.com/google/btree v1.0.0 // indirect
	github.com/gorilla/handlers v1.5.1 // indirect
	github.com/gorilla/mux v1.8.0 // indirect
	github.com/gorilla/websocket v1.4.2 // indirect
	github.com/grpc-ecosystem/go-grpc-middleware v1.3.0 // indirect
	github.com/grpc-ecosystem/grpc-gateway v1.16.0 // indirect
	github.com/gsterjov/go-libsecret v0.0.0-20161001094733-a6f4afe4910c // indirect
	github.com/gtank/merlin v0.1.1 // indirect
	github.com/gtank/ristretto255 v0.1.2 // indirect
	github.com/hashicorp/go-immutable-radix v1.0.0 // indirect
	github.com/hashicorp/golang-lru v0.5.4 // indirect
	github.com/hashicorp/hcl v1.0.0 // indirect
	github.com/hdevalence/ed25519consensus v0.0.0-20210204194344-59a8610d2b87 // indirect
	github.com/improbable-eng/grpc-web v0.14.1 // indirect
	github.com/inconshreveable/mousetrap v1.0.0 // indirect
	github.com/interchainio/tm-load-test v0.9.0 // indirect
	github.com/jmhodges/levigo v1.0.0 // indirect
	github.com/keybase/go-keychain v0.0.0-20190712205309-48d3d31d256d // indirect
	github.com/libp2p/go-buffer-pool v0.0.2 // indirect
	github.com/magiconair/properties v1.8.5 // indirect
	github.com/mattn/go-isatty v0.0.14 // indirect
	github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
	github.com/mimoo/StrobeGo v0.0.0-20181016162300-f8f6d4d2b643 // indirect
	github.com/mitchellh/mapstructure v1.4.2 // indirect
	github.com/mtibben/percent v0.2.1 // indirect
	github.com/pelletier/go-toml v1.9.4 // indirect
	github.com/petermattis/goid v0.0.0-20180202154549-b0b1615b78e5 // indirect
	github.com/pkg/errors v0.9.1 // indirect
	github.com/pmezard/go-difflib v1.0.0 // indirect
	github.com/prometheus/client_golang v1.11.0 // indirect
	github.com/prometheus/client_model v0.2.0 // indirect
	github.com/prometheus/common v0.30.0 // indirect
	github.com/prometheus/procfs v0.6.0 // indirect
	github.com/rakyll/statik v0.1.7 // indirect
	github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0 // indirect
	github.com/regen-network/cosmos-proto v0.3.1 // indirect
	github.com/rs/zerolog v1.25.0 // indirect
	github.com/sasha-s/go-deadlock v0.2.1-0.20190427202633-1595213edefa // indirect
	github.com/spf13/afero v1.6.0 // indirect
	github.com/spf13/cast v1.4.1 // indirect
	github.com/spf13/cobra v1.2.1 // indirect
	github.com/spf13/jwalterweatherman v1.1.0 // indirect
	github.com/spf13/pflag v1.0.5 // indirect
	github.com/spf13/viper v1.9.0 // indirect
	github.com/stretchr/testify v1.7.0 // indirect
	github.com/subosito/gotenv v1.2.0 // indirect
	github.com/syndtr/goleveldb v1.0.1-0.20200815110645-5c35d600f0ca // indirect
	github.com/tecbot/gorocksdb v0.0.0-20191217155057-f0fad39f321c // indirect
	github.com/tendermint/btcd v0.1.1 // indirect
	github.com/tendermint/crypto v0.0.0-20191022145703-50d29ede1e15 // indirect
	github.com/tendermint/go-amino v0.16.0 // indirect
	github.com/tendermint/tendermint v0.34.13 // indirect
	github.com/tendermint/tm-db v0.6.4 // indirect
	github.com/zondax/hid v0.9.0 // indirect
	go.etcd.io/bbolt v1.3.5 // indirect
	golang.org/x/crypto v0.0.0-20210817164053-32db794688a5 // indirect
	golang.org/x/net v0.0.0-20210525063256-abc453219eb5 // indirect
	golang.org/x/sys v0.0.0-20210823070655-63515b42dcdf // indirect
	golang.org/x/term v0.0.0-20201126162022-7de9c90e9dd1 // indirect
	golang.org/x/text v0.3.6 // indirect
	google.golang.org/genproto v0.0.0-20210828152312-66f60bf46e71 // indirect
	google.golang.org/grpc v1.40.0 // indirect
	google.golang.org/protobuf v1.27.1 // indirect
	gopkg.in/ini.v1 v1.63.2 // indirect
	gopkg.in/yaml.v2 v2.4.0 // indirect
	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
)

replace google.golang.org/grpc => google.golang.org/grpc v1.33.2

replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1

replace github.com/tendermint/tendermint => github.com/tendermint/tendermint v0.34.13

replace github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
```

**Comment by marbar3778** [comment]
this looks correct. What version of tendermint or cosmos-sdk are you upgrading from?

**Comment by puneetsingh166** [comment]
when i am trying to install this package  ``` go get github.com/interchainio/tm-load-test/pkg/loadtest ``` over it . then these error is coming
 ```  github.com/you/my-load-tester/cmd/my-load-tester imports
       github.com/interchainio/tm-load-test/pkg/loadtest imports
       github.com/tendermint/tendermint/libs/common: module github.com/tendermint/tendermint@latest found (v0.34.13, replaced by github.com/tendermint/tendermint@v0.34.13), but does not contain package github.com/tendermint/tendermint/libs/common
github.com/you/my-load-tester/cmd/my-load-tester imports
       github.com/interchainio/tm-load-test/pkg/loadtest imports
       github.com/tendermint/tendermint/rpc/lib/types: module github.com/tendermint/tendermint@latest found (v0.34.13, replaced by github.com/tendermint/tendermint@v0.34.13), but does not contain package github.com/tendermint/tendermint/rpc/lib/types ```
``` 
other than this  running smooth .

**Comment by marbar3778** [comment]
>   github.com/interchainio/tm-load-test/pkg/loadtest

Ohhh, use   github.com/interchainio/tm-load-test/pkg/loadtest, the repo moved.

**Comment by puneetsingh166** [comment]
downloaded the git repo locally using ``` go get github.com/informalsystems/tm-load-test ``` then trying to import it  like this in main.go file  ``` github.com/interchainio/tm-load-test/pkg/loadtest ``` 
then  getting error 
```  github.com/interchainio/tm-load-test/pkg/loadtest
..\..\..\go\src\github.com\interchainio\tm-load-test\pkg\loadtest\slave.go:324:38: multiple-value uuid.NewV4() in single-value context
..\..\..\go\src\github.com\interchainio\tm-load-test\pkg\loadtest\tm_network_info.go:69:26: too many arguments in call to "github.com/tendermint/tendermint/rpc/client/http".New
        have (string, string)
        want (string)
..\..\..\go\src\github.com\interchainio\tm-load-test\pkg\loadtest\tm_network_info.go:145:74: peerInfo.RemoteIP undefined (type coretypes.Peer has no field or method RemoteIP)
..\..\..\go\src\github.com\interchainio\tm-load-test\pkg\loadtest\tm_network_info.go:189:27: too many arguments in call to "github.com/tendermint/tendermint/rpc/client/http".New
        have (string, string)
        want (string)
``` 

before it @marbar3778 let me explain what i am trying to do. basically i am writing [my custom client](https://github.com/informalsystems/tm-load-test/blob/master/pkg/loadtest/README.md ) using tm-load-test for cosmos node.  so trying to import all these below packages in client.go [./pkg/myabciapp/client.go](https://github.com/informalsystems/tm-load-test/blob/master/pkg/loadtest/README.md)
``` 
 package myabciapp

import (
    "github.com/informalsystems/tm-load-test"
    "fmt"
    banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
    "github.com/cosmos/cosmos-sdk/simapp"
    "github.com/cosmos/cosmos-sdk/testutil/testdata"
    "github.com/cosmos/cosmos-sdk/types"

    // cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
	"github.com/cosmos/cosmos-sdk/types/tx/signing"
	xauthsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
    "github.com/cosmos/cosmos-sdk/client/tx"
)
``` 

here my client.go file 
``` package myabciapp

import (
    "github.com/interchainio/tm-load-test/pkg/loadtest"
    "fmt"
    banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
    "github.com/cosmos/cosmos-sdk/simapp"
    "github.com/cosmos/cosmos-sdk/testutil/testdata"
    "github.com/cosmos/cosmos-sdk/types"

    // cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
	"github.com/cosmos/cosmos-sdk/types/tx/signing"
	xauthsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
    "github.com/cosmos/cosmos-sdk/client/tx"
)


// MyABCIAppClientFactory creates instances of MyABCIAppClient
type MyABCIAppClientFactory struct {}

// MyABCIAppClientFactory implements loadtest.ClientFactory
var _ loadtest.ClientFactory = (*MyABCIAppClientFactory)(nil)

// MyABCIAppClient is responsible for generating transactions. Only one client
// will be created per connection to the remote Tendermint RPC endpoint, and
// each client will be responsible for maintaining its own state in a
// thread-safe manner.
type MyABCIAppClient struct {}

// MyABCIAppClient implements loadtest.Client
var _ loadtest.Client = (*MyABCIAppClient)(nil)

func (f *MyABCIAppClientFactory) ValidateConfig(cfg loadtest.Config) error {
    // Do any checks here that you need to ensure that the load test 
    // configuration is compatible with your client.
    return nil
}

func (f *MyABCIAppClientFactory) NewClient(cfg loadtest.Config) (loadtest.Client, error) {
    return &MyABCIAppClient{}, nil
}

// GenerateTx must return the raw bytes that make up the transaction for your
// ABCI app. The conversion to base64 will automatically be handled by the 
// loadtest package, so don't worry about that. Only return an error here if you
// want to completely fail the entire load test operation.
func (c *MyABCIAppClient) GenerateTx() ([]byte, error) {
    
// 	
      // following function.
      encCfg := simapp.MakeTestEncodingConfig()

      // Create a new TxBuilder.
      txBuilder := encCfg.TxConfig.NewTxBuilder()
    // --snip--

    priv1, _, addr1 := testdata.KeyTestPubAddr()
    // priv2, _, addr2 := testdata.KeyTestPubAddr()
     priv3, _, addr3 := testdata.KeyTestPubAddr()
    // fmt.Println(priv1)
    //  fmt.Println(priv2)
      fmt.Println(priv3)

    // Define two x/bank MsgSend messages:
    // - from addr1 to addr3,
    // - from addr2 to addr3.
    // This means that the transactions needs two signers: addr1 and addr2.
    msg1 := banktypes.NewMsgSend(addr1, addr3, types.NewCoins(types.NewInt64Coin("atom", 12)))
    // msg2 := banktypes.NewMsgSend(addr2, addr3, types.NewCoins(types.NewInt64Coin("atom", 34)))

    err := txBuilder.SetMsgs(msg1)
    if err != nil {
       fmt.Println("error")
    }

    txBuilder.SetGasLimit(20)
    txBuilder.SetFeeAmount(types.NewCoins(types.NewInt64Coin("atom", 12)))
    txBuilder.SetMemo("50stake")
    txBuilder.SetTimeoutHeight(50)


    //-------------


    // privs := []cryptotypes.PrivKey{priv1, priv2}
    // accNums:= []uint64{0} // The accounts' account numbers
    // accSeqs:= []uint64{0} // The accounts' sequence numbers

    // First round: we gather all the signer infos. We use the "set empty
    // signature" hack to do that.
    //  var sigsV2 []signing.SignatureV2
    //  for i, priv := range privs {
        sigV2 := signing.SignatureV2{
            PubKey: priv1.PubKey(),
            Data: &signing.SingleSignatureData{
                SignMode:  encCfg.TxConfig.SignModeHandler().DefaultMode(),
                Signature: nil,
            },
            Sequence: 0,
        }
     
        // fmt.Println("1234567")

        // fmt.Println("sigsV2", sigsV2)
        // fmt.Println("sigV2", sigV2)

        //  sigsV2 = append(sigsV2, sigV2)
    //  }
    // fmt.Println("nsssssssssssssssss")

    err = txBuilder.SetSignatures(sigV2)
    if err != nil {
       fmt.Println("error")
    }

    // Second round: all signer infos are set, so each signer can sign.
    // sigsV2 = []signing.SignatureV2{}
    //  for i, priv := range privs {
        signerData := xauthsigning.SignerData{
            ChainID:       "onomyd",
            AccountNumber: 0,
            Sequence:      0,
        }
        sigV2, err = tx.SignWithPrivKey(
            encCfg.TxConfig.SignModeHandler().DefaultMode(), signerData,
            txBuilder, priv1, encCfg.TxConfig, 0)
        if err != nil {
            fmt.Println("error")
        }

        //  sigsV2 = append(sigsV2, sigV2)
    //  }
    err = txBuilder.SetSignatures(sigV2)
    if err != nil {
        fmt.Println("error")
    }

txBytes, err := encCfg.TxConfig.TxEncoder()(txBuilder.GetTx())
if err != nil {
    fmt.Println("na")
}

// fmt.Println("fmt",txBytes)
return txBytes, nil
}
```
for cosmos node. 
after it when running ```go mod tidy``` command then that above error coming of tendermint. my be some version issue. i seem tm-load- testing using different version of tendermint and cosmos SDK using some other version of tendermint. if i am not wrong.

**Comment by marbar3778** [comment]
the errors you are seeing is correct. You must fix the issues in order to make the build pass.

**Comment by creachadair** [comment]
I believe as @marbar3778 said v0.34.13 of tendermint should be OK here. I looked into the build errors you describe here, and I think they should be fairly easy to resolve.

Besides updating to the new location for the tm-load-test repository, I found I also needed to clone one of the protobuf replacements that cosmos-sdk depends on, in order to get your example to build:

```
go get github.com/informalsystems/tm-load-test
go mod edit -replace 'github.com/gogo/protobuf=github.com/regen-network/protobuf@v1.3.3-alpha.regen.1'
```

The `go` tool will also complain about the import/module mismatch on the tm-load-test repo. For testing, I replaced it locally, and filed https://github.com/informalsystems/tm-load-test/pull/108 to update the module name there.

With those changes, however, I was able to at least _build_ the code you have here. I don't know if it actually works in production, but I hope that may at least help you get unstuck.
